packages/react-dom/src/InView.tsx

```diff
@@ -2,8 +2,26 @@ import { type InViewOptions, useInView } from './useInView'
 
 interface InViewProps extends InViewOptions {
   children: (inViewResult: ReturnType<typeof useInView>) => React.ReactNode
+  onInView?: (entry: IntersectionObserverEntry) => void
+  onInViewEnd?: (entry: IntersectionObserverEntry) => void
 }
 
 export function InView({ children, ...options }: InViewProps) {
-  return <>{children(useInView(options))}</>
+  return (
+    <>
+      {children(
+        useInView({
+          ...options,
+          onChange: (inView, entry) => {
+            options.onChange?.(inView, entry)
+            if (inView) {
+              options.onInView?.(entry)
+            } else {
+              options.onInViewEnd?.(entry)
+            }
+          },
```

Comment:
Wouldn’t onInViewEnd be triggered when the InView component gets rendered but isn’t in the viewport? 👀

Reply:
When I test in visualization, You're right. 
I tried many things but it doesn't work..! If you have any idea what to do, please help me.

Reply:
@manudeli How about this?

```ts
export function InView({ children, ...options }: InViewProps) {
  const hasBeenInViewRef = useRef(false)

  return (
    <>
      {children(
        useInView({
          ...options,
          onChange: (inView, entry) => {
            options.onChange?.(inView, entry)
            if (inView) {
              hasBeenInViewRef.current = true
              options.onInView?.(entry)
            } else if (hasBeenInViewRef.current) {
              options.onInViewEnd?.(entry)
            }
          },
        })
      )}
    </>
  )
}
```

Reply:
You're king 👑 I use your way in [d9274e4](https://github.com/toss/suspensive/pull/1518/commits/d9274e456a48db711ea0ad6a6d211ca7e20c951c)